sw/device/tests/rv_core_ibex_nmi_irq_test.c

```diff
@@ -107,52 +93,59 @@ static void alert_handler_config(void) {
 
   alert_handler_testutils_configure_all(&alert_handler, config,
                                         /*lock=*/kDifToggleDisabled);
-  // Enables alert handler irq.
-  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
-      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
 }
 
 /**
- * Handle the wdog bark nmi irq.
+ * Handle the wdog bark NMI irq.
  */
 static void wdog_handler(void) {
-  top_earlgrey_plic_peripheral_t peripheral;
-  dif_aon_timer_irq_t irq;
-  isr_testutils_aon_timer_isr(plic_ctx, aon_timer_ctx, &peripheral, &irq);
+  bool is_pending;
+  // The watchdog bark external interrupt is also connected to the NMI input
+  // of rv_core_ibex. We therefore expect the interrupt to be pending on the
+  // peripheral side.
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      &aon_timer, kDifAonTimerIrqWdogTimerBark, &is_pending));
+  CHECK(is_pending, "Expected watchdog bark interrupt to be pending");
```

Comment:
This PR looks really good, the only minor thing I would change is moving this `CHECK` out of the irq handler (by storing the state in a global) to avoid issues or rework when using preeption. The same for the alert_handler_handler.

Reply:
Thanks @engdoreis! I have changed this so that the state is assigned to a global, and moved the check to the corresponding test functions called in main.